[TRCL-409][TRCL-410][TRCL-411][feat] Introduce more streak-cam automatic protection

pieleric opened the pull request:
Improves the automatic protection of the streak-cam in various ways:
* Pause key also protects the streak-cam
* Protection not only sets the MCP Gain to 0, but also closes the shutter, if available
* The change of the wavelength or e-beam blanker also trigger the protection
* In the SPARC alignment tab, the "dangerous" settings are always reset when going back to the tab.

pieleric committed:
[fix] Adjust level of the notification message to show hardware errors on Ubuntu 22.04

On Ubuntu 22.04, Gnome doesn't show on the screen the notifications
of level "info" (or below). They are only shown if the user looks at the
notification tray.

Errors with the microscope hardware are typically very important (on a computer
whose main purpose is to control the hardware). So raise the errors to
"ERROR" level, and when an error is recovered, to warning.
This way, all the user should notice these errors better.

pieleric committed:
[feat] Pause key also set detectors to "safe mode"

For now the [Pause] key (and "Stop all axes" menu) would only stop the
movement of the actuators, in order to bring the hardware to a safe
state. However, we can do more than that. Some detectors can be put a
safe state, to handle cases where too much "energy" is received. For
instance, the streak-cam has a shutter and gain which can be adjusted
for safety.

=> Introduce a new generic function to place "all detector" to a safe
mode... and for now add support for one: the streak-unit.

Also add a sort of callback to the tabs, to know when this function was
triggered. Will be used in a futher commit to handle better GUI updates
in such case.

pieleric committed:
[feat] Protect streak-cam when unblanking e-beam

When the e-beam is unblanked (even when going from pulsed-blanked to
unblanked), the sample may generate a lot more light. This can be
dangerous for the streak-cam.

=> If this happens, protect the streak-cam.

Also extend the sparc2 tabs to listen to the "hardware protection"
request by the user, and pause the streams in such case, for extra
safety (and keeping the stream VA updates simple).

pieleric committed:
[feat] Extend streak-cam live streams to protect the camera in more cases

The streak-cam streams already support some protection.
If the camera supports shutter, it should be closed too.

If the center wavelength goes from a grating to 0th order, the
protection should be triggered too.

Also add some test cases for the StreakCamStream.

pieleric commented on src/odemis/acq/stream/_helper.py:
```diff
@@ -870,31 +879,53 @@ def _is_active_setter(self, active):
                     self.streak_unit.MCPGain.value = 0
+                    # TODO: Is it good for the shutter to be repeatedly opened/closed whenever playing/stopping?
+                    if model.hasVA(self.streak_unit, "shutter"):
+                        self.streak_unit.shutter.value = True
```
`shutter = True`  means the shutter gets closed (and stays so) . So, no light anymore.

tmoerkerken commented on src/odemis/gui/cont/tabs/sparc2_align_tab.py:
```diff
@@ -2011,6 +2020,48 @@ def _onFiberPos(self, pos):
+
+        if protected:
+            popup.show_message(self.main_frame, "Streak camera protection",
+                               message="Reset streak camera settings were reset due "
```
```suggestion
                               message="Streak camera settings were reset due "
```

pieleric committed:
[fix] TemporalSpectrum acquisition should protect streakcam after acquisition

Live stream took care of protecting the streakcam, but not the
MD acquisition stream.
=> Let the MD stream ask the Live stream to protect the streakcam at
the end of the acquisition (to avoid duplicating code).
